src/segment-loader.js

```diff
@@ -128,6 +149,13 @@ const segmentInfoString = (segmentInfo) => {
   ].join(' ');
 };
 
+const isFmp4 = (segment) => {
+  // TODO we need a better way if determining if a segment is an mp4 or not. We rely on
+  // this method at the moment because the spec requires a map tag on each fmp4
+  // segment), however, ts segments may have the map tag as well
+  return !!segment.map;
```

Comment:
The easiest way here is to look at the first 8 bytes of the segment which will have 4 bytes for the size of the first atom (that you can ignore) followed by the four bytes: `[0x66, 0x74, 0x79, 0x70]` for the `ftyp` atom.